pkgs/development/python-modules/yq/default.nix

```diff
@@ -9,31 +10,40 @@
 , flake8
 , jq
 , pytest
+, unixtools
 , toml
 }:
 
-buildPythonApplication rec {
+buildPythonPackage rec {
   pname = "yq";
-  version = "2.10.1";
+  version = "2.11.0";
 
-  propagatedBuildInputs = [ pyyaml xmltodict jq argcomplete ];
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1gp9q5w1bjbw7wmba5hm8ippwvkind0p02n07fqa9jlqglhxhm46";
+  };
+
+  propagatedBuildInputs = [
+    pyyaml
+    xmltodict
+    argcomplete
+  ];
 
   doCheck = true;
 
   checkInputs = [
+   unixtools.script
    pytest
    coverage
    flake8
-   jq
+   pkgs.jq
    toml
   ];
 
-  checkPhase = "pytest ./test/test.py";
+  # tests fails if stdin is not a tty
+  checkPhase = "echo | script -c 'pytest ./test/test.py'";
```

Comment:
Patch no longer required.

Reply:
disgusting, but beautiful at the same time lol

Reply:
Unfortunately, the BSD [script(1)](https://www.freebsd.org/cgi/man.cgi?script(1)) doesn't have the `-c` option causing this to fail on macOS. Probably something like the following is required on macOS:

```
checkPhase = "echo | script typescript pytest ./test/test.py";
```